src/main/java/io/confluent/connect/elasticsearch/ElasticsearchSinkTask.java

```diff
@@ -15,152 +15,191 @@
-      if (maxRetryBackoffMs > RetryUtil.MAX_RETRY_TIME_MS) {
-        log.warn("This connector uses exponential backoff with jitter for retries, "
-                + "and using '{}={}' and '{}={}' results in an impractical but possible maximum "
-                + "backoff time greater than {} hours.",
-            ElasticsearchSinkConnectorConfig.MAX_RETRIES_CONFIG, config.maxRetries(),
-            ElasticsearchSinkConnectorConfig.RETRY_BACKOFF_MS_CONFIG, config.retryBackoffMs(),
-            TimeUnit.MILLISECONDS.toHours(maxRetryBackoffMs));
+      if (context.errantRecordReporter() == null) {
+        log.info("Errant record reporter not configured.");
       }
 
-      if (client != null) {
-        this.client = client;
-      } else {
-        this.client = new JestElasticsearchClient(props);
-      }
+      // may be null if DLQ not enabled
+      reporter = context.errantRecordReporter();
+    } catch (NoClassDefFoundError | NoSuchMethodError e) {
+      // Will occur in Connect runtimes earlier than 2.6
+      log.warn("AK versions prior to 2.6 do not support the errant record reporter.");
+    }
+
+    this.client = client != null ? client : new ElasticsearchClient(config, reporter);
 
-      ElasticsearchWriter.Builder builder = new ElasticsearchWriter.Builder(this.client)
-          .setType(config.type())
-          .setIgnoreKey(config.ignoreKey(), config.ignoreKeyTopics())
-          .setIgnoreSchema(config.ignoreSchema(), config.ignoreSchemaTopics())
-          .setCompactMapEntries(config.useCompactMapEntries())
-          .setFlushTimoutMs(config.flushTimeoutMs())
-          .setMaxBufferedRecords(config.maxBufferedRecords())
-          .setMaxInFlightRequests(config.maxInFlightRequests())
-          .setBatchSize(config.batchSize())
-          .setLingerMs(config.lingerMs())
-          .setRetryBackoffMs(config.retryBackoffMs())
-          .setMaxRetry(config.maxRetries())
-          .setDropInvalidMessage(config.dropInvalidMessage())
-          .setBehaviorOnNullValues(config.behaviorOnNullValues())
-          .setBehaviorOnMalformedDoc(config.behaviorOnMalformedDoc());
-
-      try {
-        if (context.errantRecordReporter() == null) {
-          log.info("Errant record reporter not configured.");
-        }
-
-        // may be null if DLQ not enabled
-        builder.setErrantRecordReporter(context.errantRecordReporter());
-      } catch (NoClassDefFoundError | NoSuchMethodError e) {
-        // Will occur in Connect runtimes earlier than 2.6
-        log.warn("AK versions prior to 2.6 do not support the errant record reporter");
+    log.info("Started ElasticsearchSinkTask.");
+  }
+
+  @Override
+  public void put(Collection<SinkRecord> records) throws ConnectException {
+    log.debug("Putting {} records to Elasticsearch.", records.size());
+    for (SinkRecord record : records) {
+      if (shouldSkipRecord(record)) {
+        log.trace("Ignoring sink {} with null value.", recordString(record));
```

Comment:
```suggestion
        log.trace("Ignoring sink record {} with null value.", recordString(record));
```

Reply:
record is generated as part of `recordString()`